cocotb/triggers.py

```diff
@@ -729,3 +729,42 @@ def _wait(self):
         for _ in range(self.num_cycles):
             yield trigger
         raise ReturnValue(self)
+
+
+@decorators.coroutine
+def with_timeout(triggers, timeout_time, timeout_unit=None):
+    """
+    Waits on triggers, throws an exception if it waits longer than the given time
+
+    Can be used with any expression that would otherwise appear to the right of a
+    `yield` (or `await` in Python 3) statement.
```

Comment:
```suggestion
    :keyword:`yield` (or :keyword:`await` in Python 3) expression.
```

Reply:
Actually, let's remove this whole sentence and put it under `triggers` below - saying the same thing twice isn't really that helpful.